### PR TITLE
Migrate db

### PR DESCRIPTION
**Pull request by halink0803**

### CODE REVIEW

**Comment by halink0803** [comment]
[Migrate user db](https://trello.com/c/V7JTf7cU/280-migrate-user-db)

File: lib/app/utils.go [hunk 1]
```diff
@@ -18,3 +19,13 @@ func joinEnvVar(prefix, envVar string) string {
 func JoinEnvVar(envVar string) string {
 	return joinEnvVar(commonEnvPrefix, envVar)
 }
+
+// BytesToUint64 converts the given bytes to a compatible uint64
+// value. It is used for reading the uint64 from a stored BoltDB
+// record. If returned 0 if the given bytes are incompatible.
+func BytesToUint64(b []byte) uint64 {
```
Comment (favadi) [inline lib/app/utils.go]: I don't think this function belong here, maybe lib/bolutil.

File: migratedb/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package main
```
Comment (favadi) [inline migratedb/cmd/main.go]: this cli should be in users/cmd/migration.

**Comment by favadi** [comment]
Please remove migratedb/cmd/dev_users.db file from db.

File: migratedb/migratedb/migrate.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+)
+
+//DBMigration user storage in bolt
+type DBMigration struct {
+	boltdb     *bolt.DB
+	postgresdb *sqlx.DB
+}
+
+//DeleteAllTables delete all table from schema using for test only
+func (dbm *DBMigration) DeleteAllTables() error {
```
Comment (favadi) [inline migratedb/migratedb/migrate.go]: unused.

File: migratedb/migratedb/migrate.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	_, err := dbm.postgresdb.Exec(fmt.Sprintf(`DROP TABLE "%s", "%s"`, addressesTableName, usersTableName))
+	return err
+}
+
+//DeleteTable delete a table from  schema
+func (dbm *DBMigration) DeleteTable(tableName string) error {
```
Comment (favadi) [inline migratedb/migratedb/migrate.go]: please make this unexported and in _test.go.

File: migratedb/migratedb/migrate.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+	_, err := dbm.postgresdb.Exec(fmt.Sprintf(`DROP TABLE "%s"`, tableName))
+	return err
+}
+
+//NewMigrateStorage connect to bolt db
+func NewMigrateStorage(dbPath string, postgres *sqlx.DB) (*DBMigration, error) {
+	var err error
+	var boltDB *bolt.DB
```
Comment (favadi) [inline migratedb/migratedb/migrate.go]: use `var(


)
`
instead of multiple var.

File: migratedb/migratedb/migrate.go [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+CREATE TABLE IF NOT EXISTS "users" (
+  id    SERIAL PRIMARY KEY,
+  email text NOT NULL UNIQUE
+);
+CREATE TABLE IF NOT EXISTS "addresses" (
+  id        SERIAL PRIMARY KEY,
+  address   text      NOT NULL UNIQUE,
+  timestamp TIMESTAMP NOT NULL DEFAULT now(),
+  user_id   SERIAL    NOT NULL REFERENCES users (id)
+);
```
Comment (favadi) [inline migratedb/migratedb/migrate.go]: duplicated from users package?

Comment (favadi) [reply migratedb/migratedb/migrate.go]: anw, don't mind, we will remove this cmd once the migration completed.

**Pull request merged**